core/arch/arm/kernel/tee_time_arm_cntpct.c

```diff
@@ -54,3 +56,39 @@ static const struct time_source arm_cntpct_time_source = {
 };
 
 REGISTER_TIME_SOURCE(arm_cntpct_time_source)
+
+/*
+ * We collect jitter using CNTPCT in 32- or 64-bit mode that is clocked at the
+ * CPU core rate.
```

Comment:
As I said twice already: `CNTPCT` is *not* clocked at the CPU core rate, it is clocked at a frequency of `CNTFRQ`.

Which raises another question now that I'm thinking about it. In low-power states, `CNTFRQ` may be lowered and `CNTPCT` incremented by more than 1 on each clock tick (to compensate for the lower update rate). See for instance the example given in the ARMv8 ARM (section G5.1.1 System counter). If the counter is incremented by steps of 500, you obviously won't be able to gather *any* entropy at all by sampling the two least significant bits.